contrib/scripts/unloaddb.sh

```diff
@@ -0,0 +1,402 @@
+         WHEN "data_type" = 'STRING' THEN prec \
+         WHEN "data_type" = 'VARCHAR' THEN prec \
+         WHEN "data_type" = 'NVARCHAR' THEN prec \
+         WHEN "data_type" = 'CHAR' THEN prec \
+         WHEN "data_type" = 'NCHAR' THEN prec \
+         WHEN "data_type" = 'TIMESTAMP' THEN 8.0 \
+         WHEN "data_type" = 'DATE' THEN 4.0 \
+         WHEN "data_type" = 'TIME' THEN 4.0 \
+         WHEN "data_type" = 'DATETIME' THEN 4.0 \
+         WHEN "data_type" = 'BIT' THEN FLOOR(prec / 8.0) \
+         WHEN "data_type" = 'BIT VARYING' THEN FLOOR(prec / 8.0) \
+         ELSE 0 \
+     END) as BIGINT)  AS [size] \
+ from db_attribute where class_name = '$table_name';"
+
+        if [ $# -eq 0 ];then
+                echo "0"
+                return
+        fi
+
+        size=$(csql $user $pass -l -c "$query" $db | grep "^<0000" | awk '{print $3}')
+        echo $size
+}
+
+function find_slot ()
+{
+        local selected=0
+        local i
+        local size=${slot_size[0]}
+
+        for ((i = 0; i < $num_proc; i++))
+        do
+                if [ ${slot_size[i]} -lt $size ];then
+                        size=${slot_size[i]}
+                        selected=$i
+                fi
+        done
+
+         echo $selected
+}
+
+function do_unloaddb ()
+{
+        local slot_num=$1
+        local i
+        local file="Tables_unloaded_$database.$slot_num"
+        local num_tables_in_slot=0
+        local msg="Success"
+        local pid
+
+        for ((i = 0; i < $num_tables; i++))
+        do
+                if [ ${slot[i]} -eq $slot_num ];then
+                        let "num_tables_in_slot++"
+                        echo "${table_selected[i]}" >> $file
+                fi
+        done
+
+        if [ $verbose = "yes" ];then
+                echo "Proc $slot_num: num tables: $num_tables_in_slot, ${slot_size[$slot_num]} bytes"
+        fi
+
+        echo "process $slot_num: starting" > unloaddb_$slot_num.status
+
+	cubrid unloaddb $user $pass --input-class-only --input-class-file $file $database &
+        pid=$!
+
+        echo $pid > unloaddb_$slot_num.pid
+
+        wait $pid
+
+        if [ $? -ne 0 ];then
+                msg="Failed"
+                echo "process $slot_num: failed" > unloaddb_$slot_num.status
+        else
+                echo "process $slot_num: success" > unloaddb_$slot_num.status
+        fi
+
+        if [ $verbose = "yes" ];then
+                echo "Finished Proc $slot_num ($msg)"
+        fi
+}
+
+function analyze_table_info ()
+{
+        local found=0
+        local line
+        local idx=0
+        local table_name
+        local db=$database
+
+        # Get all table names from CATALOG
+
+        if [ X$from_file != X"" ];then  # Read table name from file
+                result=$(cat $from_file)
+        else
+                result=$(csql $user $pass -c "select class_name from db_class where is_system_class = 'NO' AND class_type = 'CLASS' order by class_name" $db)
```

Comment:
You can use "show tables" instead of this query.
And you can also remove column headers with --skip-column-names at csql.

Reply:
Comment accepted, I changed the query to get all table names to "show tables" (it is quite simple, thank you).
In the for_loop to get table name in csql, I did
1. skip line until ===
2. remove single quota '**code**' to **code**
3. loop until no single quota is found

And if it is from file, I assumed there is one table name per line. (no single quota)